Makefile

```diff
@@ -1,59 +1,59 @@
-sensirion_common_dir := ./embedded-common
-CFLAGS := -Wall -Werror -I. -I${sensirion_common_dir}
+.PHONY: FORCE all
 
-common_objects := git_version.o sht_common.o example_usage.o
-sw_objects := ${sensirion_common_dir}/sw_i2c/sensirion_sw_i2c.o ${sensirion_common_dir}/sw_i2c/sensirion_sw_i2c_implementation.o
-hw_objects := ${sensirion_common_dir}/hw_i2c/sensirion_hw_i2c_implementation.o
-all_objects := ${common_objects} ${hw_objects} ${sw_objects} shtc1.o sht3x.o git_version.c
+all: sht3x shtc1
 
-binaries := example_usage_shtc1_sw example_usage_shtc1_hw \
-            example_usage_sht3x_sw example_usage_sht3x_hw
+sht3x/sht3x.o: sht-common/sht_common.h sht3x/sht3x.c
+	cd sht3x && make
```

Comment:
I don't like how we have to specify the source dependency for the subdirectory twice, here and in the subdirectory Makefile itself.

Reply:
A quick search turned up this: http://owen.sj.ca.us/~rk/howto/slides/make/slides/makerecurs.html
The easiest solutions seems to be to just call the subdirectory Makefile always.
Also we should probably pass the `$(MFLAGS)` to the make call.

Reply:
True, it's probably better (less error prone in the future) to always call make to recurse, even if it only then finds that there's nothing to do. I didn't know about MFLAGS, will add.